docs/app/pages/components/components-sections/input-controls/typeahead/snippets/app.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, ElementRef, HostListener, Inject } from '@angular/core';
+import 'chance';
+import { TypeaheadOptionEvent } from '@ux-aspects/ux-aspects';
+
+@Component({
+    selector: 'app',
+    templateUrl: 'app.component.html',
+    styleUrls: ['./app.component.css'],
+})
+export class AppComponent {
+
+    tagDocumentationRoute: string;
+    values: string[] = [];
```

Comment:
When using arrays in HTML we always try to use `ReadonlyArray` instead of a normal array. This forces the reference to change each time the array changes and can improve performance in large applications so we should promote this good practice in our sample code.